### PR TITLE
Refactor something

### PR DESCRIPTION
**Pull request by ketal178**
# Refactor something

## 更改内容 / Types of Changes
- 增强优化 Enhancement/optimization

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  

## 检查列表 / Checklist
- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [x] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/me/ketal/hook/SortAtPanel.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        try {
+            uin!!.toLong()
+        } catch (e: Exception) {
+            return null
+        }
+        return uin
+    }
+
+    override fun isValid(): Boolean {
+        return (H.isTIM() && H.getVersionCode() >= TIMVersion.TIM_3_1_1) or !H.isTIM() && H.getVersionCode() >= QQVersion.QQ_8_1_3
```
Comment (singleNeuron) [inline app/src/main/java/me/ketal/hook/SortAtPanel.kt:88]: 都kotlin了还在这用Java的wrapper呢，hostinformationProvider.isTim不香？

**Review by singleNeuron — CHANGES_REQUESTED** [review]

File: app/src/main/java/me/ketal/hook/SortAtPanel.kt (hunk 1 above)
Comment (ketal178) [reply app/src/main/java/me/ketal/hook/SortAtPanel.kt:88]: 重构=Java2Kotlin Converter

Comment (cinit) [reply app/src/main/java/me/ketal/hook/SortAtPanel.kt:88]: 哎呀， 不要xx开发者

Comment (ketal178) [reply app/src/main/java/me/ketal/hook/SortAtPanel.kt:88]: 我感觉H还短些
代码能短就短

Comment (cinit) [reply app/src/main/java/me/ketal/hook/SortAtPanel.kt:88]: @singleNeuron  长度，谢谢

Comment (singleNeuron) [reply app/src/main/java/me/ketal/hook/SortAtPanel.kt:88]: > 我感觉H还短些
> 代码能短就短

可读性不要了？

**Review by cinit — APPROVED** [review]

**Review by singleNeuron — APPROVED** [review]

**Review by NextAlone — DISMISSED** [review]

**Review by cinit — DISMISSED** [review]

**Review by singleNeuron — DISMISSED** [review]

**Review by NextAlone — DISMISSED** [review]

**Review by NextAlone — APPROVED** [review]

**Review by qwq233 — DISMISSED** [review]
refactor something到底refactor了个什么

**Comment by cinit** [comment]
> refactor something到底refactor了个什么

把78长的 `HostInformationProviderKt.getHostInformationProvider().getApplicationContext()` 简化到了56长的`HostInformationProviderKt.getHostInfo().getApplication()`

**Comment by qwq233** [comment]
@cinit 哎呀说具体点嘛

**Comment by singleNeuron** [comment]
@cinit 谁让你不用Kotlin

**Comment by qwq233** [comment]
@cinit commit message说清楚

**Comment by singleNeuron** [comment]
@cinit

**Review by singleNeuron — APPROVED** [review]

**Review by NextAlone — APPROVED** [review]

**Review by cinit — APPROVED** [review]

**Pull request merged**